File: csharp/Betfair.ESAClient/Betfair.ESAClient/Cache/MarketSnap.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Betfair.ESASwagger.Model;

namespace Betfair.ESAClient.Cache
{
    /// <summary>
    /// Thread safe atomic snapshot of a market.
    /// Reference only changes if the snapshot changes:
    /// i.e. if snap1 == snap2 then they are the same.
    /// (same is true for sub-objects)
    /// </summary>
    public class MarketSnap
    {
        public MarketDefinition MarketDefinition { get; internal set; }
        public string MarketId { get; internal set; }
        public IList<MarketRunnerSnap> MarketRunners { get; internal set; }
        public double TradedVolume { get; internal set; }

        public override string ToString()
        {
            return "MarketSnap{" +
                "MarketId=" + MarketId +
                ", MarketDefinition=" + MarketDefinition +
                ", MarketRunners=" + String.Join(", ", MarketRunners) +
                ", TradedVolume=" + TradedVolume +
                "}";
        }
    }
}

File: java/client/src/main/java/com/betfair/esa/client/cache/market/MarketSnap.java
package com.betfair.esa.client.cache.market;

import com.betfair.esa.swagger.model.MarketDefinition;
import java.util.List;

/**
 * Thread safe atomic snapshot of a market. Reference only changes if the snapshot changes: i.e. if
 * snap1 == snap2 then they are the same (same is true for sub-objects)
 */
public class MarketSnap {
    private String marketId;
    private MarketDefinition marketDefinition;
    private List<MarketRunnerSnap> marketRunners;
    private double tradedVolume;

    public String getMarketId() {
        return marketId;
    }

    void setMarketId(String marketId) {
        this.marketId = marketId;
    }

    public MarketDefinition getMarketDefinition() {
        return marketDefinition;
    }

    void setMarketDefinition(MarketDefinition marketDefinition) {
        this.marketDefinition = marketDefinition;
    }

    public List<MarketRunnerSnap> getMarketRunners() {
        return marketRunners;
    }

    void setMarketRunners(List<MarketRunnerSnap> marketRunners) {
        this.marketRunners = marketRunners;
    }

    public double getTradedVolume() {
        return tradedVolume;
    }

    void setTradedVolume(double tradedVolume) {
        this.tradedVolume = tradedVolume;
    }

    @Override
    public String toString() {
        return "MarketSnap{"
                + "marketId='"
                + marketId
                + '\''
                + ", marketDefinition="
                + marketDefinition
                + ", marketRunners="
                + marketRunners
                + ", tradedVolume="
                + tradedVolume
                + '}';
    }
}
